### PR TITLE
Dt 1267 search by ldu code

### PR DESCRIPTION
**Pull request by rgforsyth**
New endpoint /ldu-codes, that takes a list of ldu codes in the body and returns offenders who have an active offender manager with a team localDeliveryUnit code matching.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt [hunk 1]
```diff
@@ -25,146 +26,172 @@ import java.util.*
...
-      }
-      matchingAllFieldsQuery
-          .mustWhenPresent("otherIds.nomsNumber", nomsNumber)
-          .mustWhenPresent("otherIds.crn", crn)
-          .mustWhenPresent("firstName", firstName)
-          .mustWhenPresent("surname", surname)
-          .mustWhenPresent("dateOfBirth", dateOfBirth)
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(this::class.java)
+        private const val MAX_SEARCH_RESULTS = 100
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt:31]: I think maybe add something to the swagger to indicate this constraint. We might need to raise it for this "bulk" end points, but not sure

Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt:31]: Ignore, so that doesn't apply to the new endpoints right?

**Review by andymarke — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt (hunk 1 above)
Comment (rgforsyth) [reply src/main/kotlin/uk/gov/justice/hmpps/offendersearch/services/SearchService.kt:31]: I set the max results for this ldu-codes endpoint (searchSourceBuilder.size) to be MAX_SEARCH_RESULTS. Is this appropriate?

**Review by andymarke — APPROVED** [review]
LGTM

**Pull request merged**